src/test/java/com/slack/kaldb/server/KaldbQueryServiceTest.java

```diff
@@ -0,0 +1,153 @@
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.grpc.GrpcService;
+import com.slack.kaldb.config.KaldbConfig;
+import com.slack.kaldb.logstore.LogMessage;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import com.slack.kaldb.proto.service.KaldbSearch;
+import com.slack.kaldb.proto.service.KaldbServiceGrpc;
+import com.slack.kaldb.testlib.ChunkManagerUtil;
+import com.slack.kaldb.testlib.KaldbConfigUtil;
+import com.slack.kaldb.testlib.MessageUtil;
+import com.slack.kaldb.testlib.TestKafkaServer;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class KaldbQueryServiceTest {
+
+  @ClassRule public static final S3MockRule S3_MOCK_RULE = S3MockRule.builder().silent().build();
+  private static SimpleMeterRegistry indexerMetricsRegistry;
+  private static ChunkManagerUtil<LogMessage> chunkManagerUtil;
+
+  private static KaldbServiceGrpc.KaldbServiceBlockingStub readServiceStub;
+  private static Server indexingServer;
+  private static Server readServer;
+
+  private static final String TEST_KAFKA_TOPIC = "test-topic";
+
+  // Kafka producer creates only a partition 0 on first message. So, set the partition to 0 always.
+  private static final int TEST_KAFKA_PARTITION = 0;
+
+  private static final String KALDB_TEST_CLIENT = "kaldb-test-client";
+  private static final String TEST_S3_BUCKET = "test-s3-bucket";
+  private static TestKafkaServer kafkaServer;
+
+  @BeforeClass
+  public static void initialize() throws Exception {
+    kafkaServer = new TestKafkaServer();
+    EphemeralKafkaBroker broker = kafkaServer.getBroker();
+    assertThat(broker.isRunning()).isTrue();
+
+    KaldbConfigs.KaldbConfig kaldbConfig =
+        KaldbConfigUtil.makeKaldbConfig(
+            "localhost:" + broker.getKafkaPort().get(),
+            8080,
+            TEST_KAFKA_TOPIC,
+            TEST_KAFKA_PARTITION,
+            KALDB_TEST_CLIENT,
+            TEST_S3_BUCKET,
+            8081);
+    KaldbConfig.initFromConfigObject(kaldbConfig);
+
+    indexingServer = newIndexingServer(kaldbConfig);
+    indexingServer.start().join();
+
+    // Produce messages to kafka, so the indexer can consume them.
+    final LocalDateTime startTime = LocalDateTime.of(2020, 10, 1, 10, 10, 0);
+    produceMessagesToKafka(broker, startTime);
+    Thread.sleep(1000); // Wait for consumer to finish consumption and roll over chunk.
+    assertThat(getCount(MESSAGES_RECEIVED_COUNTER, indexerMetricsRegistry)).isEqualTo(100);
+
+    readServer = newReadServer(kaldbConfig);
+    readServer.start().join();
+
+    // We want to query the indexing server
+    List<String> servers = new ArrayList<>();
+    servers.add("gproto+http://127.0.0.1:8080/");
+    KaldbQueryService.servers = servers;
+    readServiceStub =
```

Comment:
I think it would also be awesome if we can assert some params on the search result.